connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala

```diff
@@ -1124,7 +1124,78 @@ class ProtobufFunctionsSuite extends QueryTest with SharedSparkSession with Prot
     }
   }
 
-  test("Corner case: empty recursive proto fields should be dropped") {
+  test("Retain empty proto fields when retain.empty.message.types=true") {
+    // When retain.empty.message.types=true, empty proto like 'message A {}' can be retained as
+    // a field by inserting a dummy column as sub column.
+    val options = Map("recursive.fields.max.depth" -> "4", "retain.empty.message.types" -> "true")
+
+    // EmptyProto at the top level. It will be an empty struct.
+    checkWithFileAndClassName("EmptyProto") {
```

Comment:
Could add the a test with recursive proto version of this test?

Reply:
There is a similar test for EmptyRecursiveProto below

Reply:
Is `EmptyRecursiveProto` used in this test?
     - If not,  lets remove this recursive setting.

Reply:
Option removed.